PR for review

ZackButcher opened the pull request.

wu-sheng committed:
Add init codes for SkyWalking probe

wu-sheng committed:
Get all attributed from adaptor.

wu-sheng committed:
Prepare codes for grpc

wu-sheng committed:
Initial version of mixer adaptor

ZackButcher commented on mixer/adapter/skywalking/grpc_test.go:
```diff
@@ -0,0 +1,37 @@
+	"fmt"
+	pb "istio.io/istio/mixer/adapter/skywalking/protocol"
+	"time"
+	"context"
```
Run `make format` https://github.com/istio/istio/wiki/Dev-Guide#auto-formatting-source-code

(or set up your editor to run [goimports](https://godoc.org/golang.org/x/tools/cmd/goimports) on save)

ZackButcher commented on mixer/adapter/skywalking/skywalking.go:
```diff
@@ -0,0 +1,161 @@
+			continue
+		}
+
+		time.Now()
```
delete

ZackButcher commented on mixer/adapter/skywalking/skywalking.go:
```diff
@@ -0,0 +1,161 @@
+
+	clientStream, err := h.client.Collect(ctx)
+	if err != nil {
+		h.env.Logger().Errorf("%v.Collect(_) = _, %v", h.client, err)
```
`return err` here since we can't send on the stream if this failed

ZackButcher commented on mixer/adapter/skywalking/skywalking.go:
```diff
@@ -0,0 +1,161 @@
+			detectPoint = pb.DetectPoint_server
+		}
+
+		var metric = pb.ServiceMeshMetric{
```
```go
metric := &pb.ServiceMeshMetric{
	// ...
}
clientStream.Send(metric)
```
instead of `var metric`; and make it a pointer rather than a value since that's what the `Send` method needs below.

ZackButcher commented on mixer/adapter/skywalking/skywalking.go:
```diff
@@ -0,0 +1,161 @@
+			DetectPoint:           detectPoint,
+		}
+
+		clientStream.Send(&metric)
```
this can return an error too, this should be:

```go
var errs error
for _, inst := range insts {
	// ...
	metric := &pb.ServiceMeshMetric{
		// ...
	}
	if err := clientStream.Send(metric); err != nil {
	 	errs = multierror.Append(errs, err)
		continue
	}
}
```

ZackButcher commented on mixer/adapter/skywalking/skywalking.go:
```diff
@@ -0,0 +1,161 @@
+	}
+
+	// Not downstream
+	_, recErr := clientStream.CloseAndRecv()
```
just call this `err`

ZackButcher commented on mixer/adapter/skywalking/skywalking.go:
```diff
@@ -0,0 +1,161 @@
+
+	// Not downstream
+	_, recErr := clientStream.CloseAndRecv()
+	if err != nil {
```
wrong var here; in this case we should inline this into the if statement (and combining with the edit I suggested above):
```go
if _, err := clientStream.CloseAndRecv(); err != nil {
	h.env.Logger().Errorf("%v.CloseAndRecv() got error %v, want %v", clientStream, recErr, nil)
	errs = multierror.Append(errs, err)
}
return errs
```

ZackButcher commented on mixer/adapter/skywalking/protocol/service-mesh.proto:
```diff
@@ -0,0 +1,59 @@
+    DetectPoint detectPoint = 16;
+}
+
+enum Protocol {
```
nit: in proto when we only use a message/enum once we declare it at its usage site:

```proto
message ServiceMeshMetric {
    // ...
    enum Protocol {
        // ...
    }
    Protocol protocol = 15;
    
    enum DetectPoint {
        // ...
    }
    DetectPoint detect_point = 15;
}
```

wu-sheng replied:
@ZackButcher How could I write the usage site? I think I miss the document about that.

ZackButcher commented on mixer/adapter/skywalking/protocol/service-mesh.proto:
```diff
@@ -0,0 +1,59 @@
+message ServiceMeshMetric {
+    int64 startTime = 1;
+    int64 endTime = 2;
+    string sourceServiceName = 3;
```
In protobufs we use `snake_case` rather than `camelCase` for field names (but message names should be `UpperCamelCase`).

wu-sheng replied:
@ZackButcher `snake_case` should be a problem. Because backend now is using the camelCase. And this proto is copied from the SkyWalking project.

wu-sheng committed:
Merge branch 'master' of https://github.com/istio/istio

zachgersh committed:
squashed perf (#7539)

ymesika committed:
Atomically create and start a test mock server (#7481)

* Atomic create and start a mock server

* Replaced NewServer(bool) with a StartNewServer()

* Removed always nil error handling

geeknoid committed:
Fix race condition in ControlZ. (#7576)

ozevren committed:
Adding an MCP backend to Mixer [Galley Integration] (#7433)

* Add a Mixer store backend for MCP protocol.

vadimeisenbergibm committed:
call the gateway container of a gateway pod "istio-proxy" (#7566)

it is required for `istioctl proxy-config` tool to work correctly. `proxy-config`
looks for "istio-proxy" container, or for a single container in a pod.

When the gateway pod contains a single gateway container, proxy-config works.
Once an additional container is added, for any purpose, proxy-config will
stop working since it will not know from which container to
dump the config information.

Calling the contaner by the same name as the deployment, as it is today,
does not add much information. Calling the container `istio-proxy` makes it
clear that we have the same proxy in the gateway pod, as the sidecar proxy in
the application pods.

sebastienvas committed:
Update codecov requirements (#7568)

Commit:
Update stale.yml (#7609)

wu-sheng committed:
Merge branch 'master' of https://github.com/istio/istio

wu-sheng committed:
Follow Zack suggestments.

bianpengyuan committed:
Update example adapter operator config to use CUMULATIVE. (#7614)

Tahler committed:
Add ShellCheck job (#7373)

* Break excludes into many lines

* Add tools/run_shellcheck.sh

* Fix for usability by alpine's ash shell

* Add "shellcheck" job which calls tools/run_shellcheck.sh

* Add comment/link to shellcheck wiki

* Add file documentation

tiswanso committed:
Prow script for pilot-multicluster-e2e test (#7376)

* Prow script for pilot-multicluster-e2e test

Runs multicluster e2e test.  Will use mason to get GKE resources in
the same GCP project.  The resource_type will need to be updated
after the test-infra commit of the boskos config change to create the new
resource_type for multiple GKE clusters.

Script can be run with pre-allocated clusters by having a pre-setup
kubeconfig for multiple clusters.  It will choose the pilot cluster to
be the first cluster in the get-contexts or the PILOT_CLUSTER env var.

example:
PILOT_CLUSTER=gke_istio-ci-testing_us-east1-c_cluster-1 \
  KUBE_USER=tiswanso@cisco.com ROOT=$(pwd) USE_MASON_RESOURCE=False \
  prow/istio-pilot-multicluster-e2e.sh

* Adress review comments and fix HUB error w/ pilot e2e

* Merge code with updated e2e-suite

-Add functions for multi-cluster setup to cluster_lib

* Make cluster_setup work for multiple clusters. Setup clusterreg_dir.

* Setup firewall rule to allow intercluster pod to pod traffic.

* Remove post rebase resolved leftover e2e-common file.

* Remove use of instance-template for finding instance tags.

gcloud IAM roles seem to have instance-template.* as a higher privilidge than instance.list|get.

* Only attempt to set firewall rules for multicluster test.

* Setup service-account and build kubeconfig for pilot use.

* Make mason_lib set default gcloud project.

* Use the correct gcloud project and exec the v1alpha3 pilot test

* e2e multicluster fix review comments

* Use different serviceaccount name to avoid new conflict with helm istio-remote

* Increase timeout for multicluster test runs.

* Use sort | uniq to built gke cidr and nettag values for fw rule

- Also, fix some variable assignment style nits

Tahler committed:
Update excludes for the shell scripts merged from release-1.0 (#7633)

Tahler committed:
ShellCheck: Enable and fix SC2006 (#7632)

* Enable check for SC2006

* Fix SC2006 errors

Commit:
Properly format values.yaml for docification (#7644)

In the istio.github.io repository, we have a tool which processes
values.yaml.  In order for that tool to work correctly, every
level in the YAML must be 2 characters exactly.  In a few cases,
the spacing isn't two, but four, or alternately the lists have a prefix
of two spaces.  I'm not sure why helm processes this incorrect YAML, but
the parser we are using in the docs repo requires strict YAML.

Tahler committed:
ShellCheck: Enable and fix syntax checking rules (#7648)

* Only check files with shebang on first line

* Fix if...then in install_prereqs_*.sh

* Enable syntax checking rules (1009, 1020, 1072, 1073)

* Fix SC2006 errors in install_prereqs_*.sh

tiswanso committed:
Fix multicluster e2e test setup to use helm chart template (#6679)

* Fix istio-remote and multicluster e2e test setup to use helm chart template

- Add istio-ingressgateway service and endpoints option to istio-remote charts
- Make the multicluster tests always setup istio-remote via helm template with
  passed args.
- Fix crash in getEndpointIPForService method

* e2e test install addons prior to remote cluster install

* Update e2e framework to work with istio-remote helm chart created configmap

- istio-remote changes remove selectorless service/endpoint setup so DNS
  for istio control plane services won't resolve on remote.  Sidecar
  injection configmap setup on remote uses the IP addresses of services
  directly.
  - change istioctl in e2e framework to allow use of sidecar inject
    configmap on remote rather than default settings.
- enable/disable sidecar injection in remote based on test flag
- misc. fix zipkin address in istio-remote configmap

* Multicluster pilot e2e: fix remote cleanup and ingress subtests

Tahler committed:
ShellCheck: Enable and fix uncommon violations (#7649)

* Enable and fix SC2043

* Enable and fix SC2196

* Enable and fix SC2039 by upgrading sh to bash

* Enable and fix SC2028

* Enable and fix SC2154

* Enable and fix SC1128

* Enable and fix SC2021 - '[[:space:]]' -> '[:space:]'

* Enable and fix SC1113

* Enable and fix SC2007

* Enable and fix SC2128

* Enable and fix SC2126

* Enable and fix SC2120

* Enable SC2119

* Enable and fix SC2103

* Enable and fix SC2220

* Enable and fix SC2209

* Enable and fix SC2002

* Update SC2043 fix

Tahler committed:
ShellCheck: Enable and fix SC2034 (#7671)

* Enable SC2034

* Remove unused '-m' flag in updateVersion.sh

* Remove unused '-i' flag in store_artifacts.sh

* Fix remaining SC2034 errors

Tahler committed:
ShellCheck: Enable and fix SC2181 (#7636)

* Enable check for SC2181

* Rewrite minikube installation

* Fix SC2181 errors

wu-sheng committed:
Add some simple documents of SkyWalking Istio probe.

wu-sheng committed:
Merge branch 'master' of https://github.com/istio/istio

wu-sheng committed:
Fix local lints and format.

wu-sheng committed:
Update license and replace the `source.service` and `destination.service`.

wu-sheng committed:
Delete proto file, to avoid license conflict with Apache licensed.
